src/openrct2-ui/scripting/CustomWindow.cpp

```diff
@@ -1055,6 +1069,21 @@ namespace OpenRCT2::Ui::Windows
                     widget.content = SCROLL_BOTH;
                 widgetList.push_back(widget);
             }
+            else if (desc.Type == "progressbar")
+            {
+                widget.type = WindowWidgetType::ProgressBar;
+                if (desc.IsDisabled)
+                {
+                    widget.colour = 1;
```

Comment:
I wasn't sure what to do for disabled state, so I went ahead and set it to white and not blinking. Some screenshots of my attempts are in discord.

Reply:
after all the feedback, decided to just remove any isDisabled functionality and leave it up to the implementer to do, whether they want to recolor, hide, etc.